ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -139,7 +148,14 @@
               true,
               false,
               ConfigDef.Importance.LOW,
-              KSQL_FUNCTIONS_SUBSTRING_LEGACY_ARGS_DOCS)
+              KSQL_FUNCTIONS_SUBSTRING_LEGACY_ARGS_DOCS),
+           new CompatibilityBreakingConfigDef(
+              KSQL_FUNCTIONS_ARRAY_LEGACY_BASE_CONFIG,
+              ConfigDef.Type.BOOLEAN,
+          true,
+          true,
```

Comment:
If you set this to false we'd get the new behavior for any new queries, without breaking anything that's already running.

Reply:
@dguy flagging this to you - what do you reckon? I've switched SUBSTRING to default to old way, but maybe switching back to default to new way, (while maintaining backwards compatibility with old interactive queries), is the way to go?

Thoughts?

Reply:
I like @rodesai 's suggestion too. Seems that @big-andy-coates also is leaning this way. Will wait for @dguy 's opinion before making any changes here.

Reply:
Sounds reasonable to me.

Reply:
OK, I'll switch SUBSTRING back to this model as well.

Reply:
SUBSTRING updated.

Reply:
So then should this be updated to `false` here?

Reply:
> So then should this be updated to false here?

Yes, I believe so. To be clear: old should be `true` and new should be `false`.